### PR TITLE
Add flattener

### PR DESCRIPTION
**Pull request by jrainville**
Works using a console command called `flatten`

Could be extended to flatten on each contract build, but that seems bit useless.

Will probably add a command later to send a request to the verify API.

Tested on Liquid pledging and everything flattened 🎉

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,13 +16,13 @@ module.exports = (embark) => {
       const [, contractNames] = cmd.split(' ');
 
       if (contractNames) {
-        embark.logger.info('Going to flatten', contractNames);
```
Comment (alaibe) [inline index.js:15]: line 11: ; // You can use `split` for commands that receive parameters
Is it a leftover from the docs?

File: lib/Flattener.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    this.embark.events.request('contracts:all', (err, contracts) => {
+      if (err) {
+        return callback(err);
+      }
+      contractNames = contractNames.split(',');
+
+      let contractsToFlatten;
+      try {
+        contractsToFlatten = contractNames.map(contractName => {
+          const contract = contracts[Object.keys(contracts).find(contractKey => contracts[contractKey].className === contractName)];
```
Comment (alaibe) [inline lib/Flattener.js]: Is Object.values available.
If it is something simpler could be written as you would directly get the contract

**Review by alaibe — COMMENTED** [review]
LGTM, however I feel that using Promise/Async/Await style would make things more easy to digest

Feel free to ignore this comment

File: index.js (hunk 1 above)
Comment (jrainville) [reply index.js:15]: Haha, yes oops.

**Comment by jrainville** [comment]
@alaibe how would you suggest doing an async/await loop inside another loop?
I legit don't know how that would be possible easily.
Also, with using the async lib with disc operations, we reach almost true parallel execution.

**Comment by alaibe** [comment]
You mean something like that?:
```
async function doSomething() {
  somethings = await AsyncCallForSomething();

  const promisesResult = somethings.map(async (value) => {
    return await AnotherAsyncCallForSomething(value);
  })

  await Promise.all(promisesResult)
}
```

My knowledge of `async` is limited but why do you think we could not achieve the same things with Promises?

With Promises you can loop in a async way too I believe

**Comment by michaelsbradleyjr** [comment]
`await Promise.all(...)` is helpful as @alaibe suggested, e.g. when you have some ops that should happen in parallel and you want to operate on the whole resolved batch. And inside an async function or async generator function you can also use `for await...of` which doesn't move to the next step in the loop until the await expressions inside the loop have resolved.

**Pull request merged**